### DISCUSSION TITLE
Transform covariates?

### DISCUSSION THREAD

**Discussion by andtheWings** [ConnorDonegan/geostan#12] [Q&A] [answered]
Hi Connor! Related question to [the other](https://github.com/ConnorDonegan/geostan/discussions/11). In the context of stangeo's [Poisson GLM](https://connordonegan.github.io/geostan/reference/stan_glm.html#details-1), is it helpful to try and normalize covariates as best as possible? My guess is yes because covariates are inside the eta parameter, which has linear relationship to log(lambda):

$$log(\lambda) = log(P) + \eta$$

If it *is* helpful to transform the covariates, should I also transform the associated error terms in the same way before feeding into `prep_me_data()`?

**Comment by ConnorDonegan** [comment, marked as answer]
If you end up transforming the scale of the covariates in any way, then the same change should be made to  the SEs. 

For MCMC sampling, it can help if they're all on a similar scale but that should not be confused with the practice of converting all the covariates to the standard normal scale by dividing by their respective standard deviations. For example, in my opinion, it makes sense, when using demographic percentages as a covariate, to use the scale 0-1 (proportions). You can also use the `center_x` argument in the models (e.g. `stan_glm`) and that can also improve sampling speed. 

When using income, for example, it helps to scale the variable down by dividing by 1,000 (this is only about MCMC sampling). So I'd divide both income and the SEs by 1000 before doing anything else, including before using `prep_me_data`.  (Its also a good idea to use log-income rather than the raw magnitude, so that makes this example a bit more complicated but similar advice applies; I use <https://connordonegan.github.io/geostan/reference/se_log.html> to create approximate SEs for the transformed variable). 

Personally i don't think that covariates need to be or should be put on the same scale (like by turning them all into z-scores), and you can probably find some valid criticisms of that practice out there.
